core/src/services/gdrive/backend.rs

```diff
@@ -0,0 +1,241 @@
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+}
+
+impl GoogleDriveBackend {
+    async fn get_abs_root_id(&self) -> String {
+        let mut req = Request::get("https://www.googleapis.com/drive/v3/files/root");
+        let auth_header_content = format!("Bearer {}", self.access_token);
+        req = req.header(header::AUTHORIZATION, auth_header_content);
+        let req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)
+            .unwrap();
+
+        let resp = self.client.send(req).await.unwrap();
+
+        let body_value: serde_json::Value =
+            serde_json::from_slice(&resp.into_body().bytes().await.unwrap()).unwrap();
+        let root_id = String::from(body_value["id"].as_str().unwrap_or_default());
+        root_id
+    }
+
+    async fn get_file_id_by_path(&self, file_path: &str) -> String {
+        let path = build_rooted_abs_path(&self.root, file_path);
+        let auth_header_content = format!("Bearer {}", self.access_token);
+
+        let mut parent_id = self.get_abs_root_id().await;
+        let file_path_items: Vec<&str> = path.split('/').filter(|&x| !x.is_empty()).collect();
+
+        for (i, item) in file_path_items.iter().enumerate() {
+            let mut query = format!(
+                "name = '{}' and parents = '{}' and trashed = false",
+                item, parent_id
+            );
+            if i != file_path_items.len() - 1 {
+                query += "and mimeType = 'application/vnd.google-apps.folder'";
+            }
+            let query: String = query.chars().filter(|c| !c.is_whitespace()).collect();
+
+            let mut req = Request::get(format!(
+                "https://www.googleapis.com/drive/v3/files?q={}",
+                query
+            ));
+            req = req.header(header::AUTHORIZATION, &auth_header_content);
+            let req = req
+                .body(AsyncBody::default())
+                .map_err(new_request_build_error)
+                .unwrap();
+
+            let resp = self.client.send(req).await.unwrap();
+
+            let body_value: serde_json::Value =
+                serde_json::from_slice(&resp.into_body().bytes().await.unwrap()).unwrap();
+            parent_id = String::from(body_value["files"][0]["id"].as_str().unwrap());
+        }
+
+        parent_id
+    }
+
+    async fn gdrive_get(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let url: String = format!(
+            "https://www.googleapis.com/drive/v3/files/{}?alt=media",
+            self.get_file_id_by_path(path).await
+        );
+
+        let auth_header_content = format!("Bearer {}", self.access_token);
+        let mut req = Request::get(&url);
+        req = req.header(header::AUTHORIZATION, auth_header_content);
+
+        let req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+
+        self.client.send(req).await
+    }
+
+    pub async fn gdrive_put(
```

Comment:
Those gdrive related API should be called like `gdrive_<native_api>`. For example, we need to use `gdrive_update` or `gdrive_create`